### PR TITLE
Idiomatic Kotlin in MockResponse.kt

### PR DESCRIPTION
**Pull request by monkey-mas**

### CODE REVIEW

File: mockwebserver/src/main/java/okhttp3/mockwebserver/Dispatcher.kt [hunk 1]
```diff
@@ -31,7 +31,8 @@ abstract class Dispatcher {
    * request bodies.
    */
   open fun peek(): MockResponse {
-    return MockResponse().setSocketPolicy(SocketPolicy.KEEP_OPEN)
+    val mockResponse = MockResponse()
+    return mockResponse.apply { mockResponse.socketPolicy = SocketPolicy.KEEP_OPEN }
```
Comment (monkey-mas) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/Dispatcher.kt]: InteliJ translated the deprecated fun to the code above.
Maybe it's better to do the following?
```
  open fun peek(): MockResponse = MockResponse().apply {
    this.socketPolicy = SocketPolicy.KEEP_OPEN
  }
```
or
```
  open fun peek(): MockResponse {
    return MockResponse().apply { this.socketPolicy = SocketPolicy.KEEP_OPEN }
  }
```

File: mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt [hunk 2]
```diff
@@ -81,8 +81,10 @@ open class QueueDispatcher : Dispatcher() {
      * Enqueued on shutdown to release threads waiting on [dispatch]. Note that this response
      * isn't transmitted because the connection is closed before this response is returned.
      */
-    private val DEAD_LETTER = MockResponse()
-        .setStatus("HTTP/1.1 $HTTP_UNAVAILABLE shutting down")
+    private val DEAD_LETTER = run {
+      val mockResponse = MockResponse()
+      mockResponse.apply { mockResponse.status = "HTTP/1.1 $HTTP_UNAVAILABLE shutting down" }
+    }
```
Comment (monkey-mas) [inline mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]: InteliJ translated the deprecated fun to the code above.
Maybe it's better to do like this?
`private val DEAD_LETTER = MockResponse().apply { this.status = "HTTP/1.1 $HTTP_UNAVAILABLE shutting down" }`

File: mockwebserver/src/main/java/okhttp3/mockwebserver/Dispatcher.kt (hunk 1 above)
Comment (swankjesse) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/Dispatcher.kt]: the 2nd one!

File: mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt (hunk 2 above)
Comment (swankjesse) [reply mockwebserver/src/main/java/okhttp3/mockwebserver/QueueDispatcher.kt]: Yes!

**Comment by swankjesse** [comment]
If you’d like to do the `apply {}` changes in a follow up, please do. Otherwise it’s fine as-is!

**Pull request merged**